lesson 1/README.md

```diff
@@ -0,0 +1,502 @@
+# Lesson 1 - Github and an Intro to Java
+
+##  Github
+
+### Ideas behind a perfect repo
+
+#### 1. What should the master branch contain?
+Working and well-tested code that has been reviewed by peers.
+
+#### 2. Non-master branches?
+1. Each of these branch should mostly have one (maybe two if you are pair-programming a feature) person working on it. 
+2. Each branch should add a significant working change to the master. Usually each branch adds a working feature to the master. Occassionally you will have a small bug that can be fixed in a few lines of code. Simply branch off master, make the change, and open a pull request to merge your fix.
+
+#### 3. Issues:
+1. Labels:
+![labels](images/labels.png)
+Every time you notice a bug, add it to the issues with the label “bug”. Whenever you/team decide to add a new features, enhancements can be used. You can use labels to prioritize certain fixed. You can also assign people to issues so that teammates don’t step into each other’s work. You can see what your teammates are working on.
+
+2. Milestones:
+![milestones](images/milestones.png)
+Use this to properly scope your project and to make sure that your team is on track.
+
+#### 4. Pull Requests:
+1. An extra checkpoint in merging a branch to master (or other branches)
+2. Peers can look at the changes a merge will add and get a chance to review/comment on/ask questions about the code
+3. Peers can (and should) checkout the branch to test it out before merging.
+
+#### 5. Merge Conflicts:
+Working in a relatively large project with multiple other people will result in merge conflicts. It happens when the same line of code has been changed on master in your branch. Here's an example of a pull request with merge conflicts:
+
+![conflict](images/conflict.png)
+
+Let's resolve the merge conflict! Try understanding what each line does from a conceptual perspective.
+
+```shell
+$ git checkout master
+Switched to branch 'master'
+Your branch is up-to-date with 'origin/master'.
+$ git pull origin master
+From github.com:davidabrahams/potential-lamp
+ * branch            master     -> FETCH_HEAD
+Already up-to-date.
+$ git checkout myBranch
+Switched to branch 'myBranch'
+$ git merge master
+Auto-merging file.java
+CONFLICT (content): Merge conflict in file.java
+Automatic merge failed; fix conflicts and then commit the result.
+```
+
+Merge conflicts might look something like this:
```

Comment:
Should we add something to here to make it clear that you're supposed to be looking in the actual file.java file in your text editor?  That was something that was weird to me when I was learning git, because everything was in the command line and then to fix stuff you actually go into the code file.